packages/rs-drive-abci/src/abci/handler/check_tx.rs

```
@@ -61,20 +66,79 @@ where
                 .cloned()
                 .unwrap_or_default();
 
+            let state_transition_name = check_tx_result
+                .state_transition_name
+                .clone()
+                .unwrap_or_else(|| "Unknown".to_string());
+
+            let priority = check_tx_result.priority as i64;
+
+            if tracing::enabled!(tracing::Level::TRACE) {
+                let message = match (r#type, code) {
+                    (0, 0) => "added to mempool".to_string(),
+                    (1, 0) => "kept in mempool after re-check".to_string(),
+                    (0, _) => format!(
+                        "rejected with code {code} due to error: {}",
+                        first_consensus_error
+                            .expect("consensus error must be present with non-zero error code")
+                    ),
+                    (1, _) => format!(
+                        "removed from mempool with code {code} after re-check due to error: {}",
+                        first_consensus_error
+                            .expect("consensus error must be present with non-zero error code")
+                    ),
+                    _ => unreachable!("we have only 2 modes of check tx"),
```

Please return an error instead.